admin/services/data-access/check-window.data.service.js

```diff
@@ -2,31 +2,76 @@
 
 const moment = require('moment')
 const CheckWindow = require('../../models/check-window')
+const winston = require('winston')
+const sqlService = require('./sql.service')
+const TYPES = require('tedious').TYPES
 
 const checkWindowDataService = {
   /**
    * Fetch check window document by id.
    * @param id
+   * @deprecated use sqlFetchCheckWindow
    * @returns {Promise.<*>}
    */
   fetchCheckWindow: async (id) => {
+    winston.warn('check-window.data.service.fetchCheckWindow is deprecated')
     return CheckWindow.findOne({'_id': id, 'isDeleted': false}).exec()
+  },
+    /**
+   * Fetch check window document by id.
+   * @param id
+   * @returns {Promise.<*>}
+   */
+  sqlFetchCheckWindow: async (id) => {
+    const sql = 'SELECT * FROM [mtc_admin].[checkWindow] WHERE isDeleted=0 AND id=@id'
+    const params = [
+      {
+        name: 'id',
+        value: id,
+        type: TYPES.Int
+      }
+    ]
+    return sqlService.query(sql, params)
```

Comment:
If this can only return at most 1 item, then the object should be returned rather than the array

Reply:
yes.  this caught me out a few times in testing